docs/develop/roadmap.md

```diff
@@ -0,0 +1,38 @@
+# roadmap
+
+The following features are considered for the near future
+
+## support none kubernetes-native controller
+
+Spiderpool support to automatically manage ippool for application, it could create, delete, scale up and down a dedicated spiderippool object with static IP address just for one application.
+
+This feature uses informer technology to watch application, parses its replicas number and manage spiderippool object, it works well with kubernetes-native controller like Deployment, ReplicaSet, StatefulSet, Job, CronJob, DaemonSet.
+
+This feature also support none kubernetes-native controller, but Spiderpool could not parse the object yaml of none kubernetes-native controller, has some limitations: 
+
+* does not support automatically scale up and down the IP
+
+* does not support automatically delete the ippool
+
+In the future, spiderpool may support all operation of automatical ippool.
+
+## support multi-cluster 
+
+* when multi-cluster use Spiderpool to assign underlay IP address in the same CIDR, spiderpool could 
+    synchronize ippool resource within a same subnet from other cluster, so it could help avoid IP conflict 
+
+* when multi-cluster use Spiderpool to manager underlay IP address, leader cluster could
+    synchronize all Spiderpool resource from member clusters, which help manager all underlay IP address
+
+## strengthen veth
+
+* support to detect IP conflict when setting up the network of pod 
+
+## integrate more CNI 
+
+* integrate more CNI addon to solve underlay network needs 
+
+## improve performance  
+
+* continually improve the performance in kinds of scenes
+
```

Comment:
Overall, the code patch does not contain any actual code changes but is instead outlining forthcoming features and improvements for a project called Spiderpool. Therefore, there are no bug risks to evaluate.

For improvement suggestions, here are a few thoughts:

- The roadmap could benefit from prioritization or scheduling information to give readers a clearer sense of what will be tackled first.
- It might also be helpful to define some terms early on (e.g., ippool, none kubernetes-native controller) to avoid confusion later.
- Improving performance is a good goal, but it would be beneficial to specify what metrics will be optimized and by how much (e.g., latency, throughput). This could help provide more specific direction and goals for developers working on that area.
- For the "strengthen veth" feature, it would be helpful to know what kind of detection mechanism will be used for IP conflicts (e.g., ARP scanning).
- Finally, for the multi-cluster support, it would be useful to understand how synchronization between clusters will occur (e.g., via APIs, custom tooling), as well as if there are any security measures in place to prevent unauthorized access.